burnman/layer.py

```diff
@@ -0,0 +1,797 @@
+from __future__ import print_function
+# This file is part of BurnMan - a thermoelastic and thermodynamic toolkit for the Earth and Planetary Sciences
+# Copyright (C) 2012 - 2017 by the BurnMan team, released under the GNU
+# GPL v2 or later.
+
+import numpy as np
+from scipy.integrate import odeint
+from scipy.integrate import quad
+from scipy.interpolate import UnivariateSpline
+from burnman import averaging_schemes
+from burnman import constants
+from burnman import geotherm
+import warnings
+
+
+from .material import Material, material_property
+from .mineral import Mineral
+from .composite import Composite
+from .seismic import Seismic1DModel
+
+import matplotlib.pyplot as plt
+
+
+class Layer(object):
+    """
+    A planetary layer class
+    """
+
+    def __init__(self, name=None, radius_planet=None,
```

Comment:
We definitely discussed this, yes. I don't know what the conclusions were. I don't think radius_planet should go in the layer init. We also risk ambiguity: is this the depth from the top of the layer, or depth from the surface? For a planet it's a different story, but I don't think many users will find it such a chore to write `depths = outer_radius - radii`